src/main/java/org/openstreetmap/atlas/checks/validation/intersections/AtGradeSignPostCheck.java

```diff
@@ -0,0 +1,634 @@
+                    .filter(outEdge -> !JunctionTag.isRoundabout(outEdge)).findFirst();
+            if (roundaboutEdge.isPresent() && roundaboutExitEdge.isPresent())
+            {
+                final Set<AtlasEntity> roundaboutEdges = this
+                        .getRoundaboutEdges((Edge) roundaboutEdge.get());
+                final AtlasEntity exitEdge = roundaboutExitEdge.get();
+                // If the destination sign relation is missing destination tag, flag it
+                if (this.isMissingDestinationTag((Edge) exitEdge))
+                {
+                    entitiesToBeFlagged.addAll(roundaboutEdges);
+                    entitiesToBeFlagged.add(roundaboutExitEdge.get());
+                }
+                final Optional<Set<AtlasEntity>> roundAboutEdgesNotPartOfRelations = this
+                        .connectedEdgesNotPartOfRelation(exitEdge, roundaboutEdges,
+                                destinationSignRelations);
+                // If there are missing destination sign relations, flag it
+                roundAboutEdgesNotPartOfRelations.ifPresent(entitiesToBeFlagged::addAll);
+            }
+        });
+        return entitiesToBeFlagged;
+    }
+
+    /**
+     * @param inEdgeToOutEdgeMap
+     *            inEdge to outEdge map
+     * @param destinationSignRelations
+     *            destinationSignRelations
+     * @return
+     */
+    private Set<AtlasEntity> getConnectedEdgesNotFormDestinationRelation(
+            final Map<AtlasEntity, Set<AtlasEntity>> inEdgeToOutEdgeMap,
+            final Set<Relation> destinationSignRelations)
+    {
+        return inEdgeToOutEdgeMap.entrySet().stream().flatMap(atlasEntitySetEntry ->
+        {
+            final AtlasEntity key = atlasEntitySetEntry.getKey();
+            return this.connectedEdgesNotPartOfRelation(key, inEdgeToOutEdgeMap.get(key),
+                    destinationSignRelations).stream();
+        }).flatMap(Collection::stream).collect(Collectors.toSet());
+    }
+
+    /**
+     * Return a FlaggedIntersection with the items in the input params and appropriate instruction
+     * index based on the input params.
+     *
+     * @param roundAboutInEdgeToOutEdgeMap
+     *            map with roundabout inEdge and outEdges
+     * @param inEdgeToOutEdgeMap
+     *            map with non-roundabout inEdge and outEdges
+     * @return FlaggedIntersection with instruction index and set of flagged items based on the
+     *         input params
+     */
+    private FlaggedIntersection getFlaggedIntersection(
+            final Map<AtlasEntity, Set<AtlasEntity>> roundAboutInEdgeToOutEdgeMap,
+            final Map<AtlasEntity, Set<AtlasEntity>> inEdgeToOutEdgeMap)
+    {
+        final Set<AtlasEntity> entitiesToBeFlagged = new HashSet<>();
+        int instructionIndex = -1;
+        // Flag all in and out edges
+        if (roundAboutInEdgeToOutEdgeMap.isEmpty() && inEdgeToOutEdgeMap != null)
```

Comment:
Is there any way we could have a nonempty ```roundAboutInEdgeToOutEdgeMap``` and ```inEdgeToOutEdgeMap```? If so there should be a case to handle flagging both

Reply:
No there will not be because populateInEdgeToOutEdgeMaps method populates either roundAboutInEdgeToOutEdgeMap or inEdgeToOutEdgeMap based on presence of roundabout edges in the output.